monkey/infection_monkey/master/ip_scanner.py

```diff
@@ -61,16 +61,33 @@ def _scan_addresses(
                 address = addresses.get_nowait()
                 logger.info(f"Scanning {address.ip}")
 
-                # TODO: Catch exceptions to prevent thread from crashing
-                ping_scan_data = self._puppet.ping(address.ip, icmp_timeout)
-                port_scan_data = self._puppet.scan_tcp_ports(address.ip, tcp_ports, tcp_timeout)
+                try:
+                    ping_scan_data = self._puppet.ping(address.ip, icmp_timeout)
+                except Exception as ex:
+                    logger.warning(f"Exception encountered when pinging {address.ip}: {str(ex)}")
+                    ping_scan_data = PingScanData(False, None)
+
+                try:
+                    port_scan_data = self._puppet.scan_tcp_ports(address.ip, tcp_ports, tcp_timeout)
+                except Exception as ex:
```

Comment:
@mssalvatore why are we doing this?

Reply:
If these raise exceptions, the thread will crash and become unusable. If we can ensure that `network_scanning.scan_tcp_ports()`, `network_scanning.ping()`, and all fingerprinters always return the appropriate data structures, then we don't need to.